### ISSUE TITLE
.order(random) Throws Error in Swift 1.2

### ISSUE THREAD

**Issue by acarr** [stephencelis/SQLite.swift#95]
After upgrading to Swift 1.2 using `.order(random)` in a query throws a compile error: 

```
Ambiguous use of 'random'
```

The query that previously worked before was:

```
let result = table.filter(exp.difficulty == difficulty.rawValue).order(random).limit(1)
```

Is there a way to reference `random` so it's not ambiguous?

**Comment by stephencelis** [comment]
Try using the module name: `.order(SQLite.random)`.

I may need to disambiguate with a new name, though.

**Comment by acarr** [comment]
Still says it's ambiguous. It shows up in the autocomplete so it seems like xcode understands that's where the method is but at build time it throws the error.

**Comment by acarr** [comment]
Finding some discussions about this issue once 1.2 came out: https://github.com/Alamofire/Alamofire/issues/410

**Comment by stephencelis** [comment]
You know what? I double-checked and `SQLite.random` is a function, not an expression, so you'll need to call it as follows:

``` swift
let result = table.filter(exp.difficulty == difficulty.rawValue).order(random()).limit(1)
```

I have a test case for this:

https://github.com/stephencelis/SQLite.swift/blob/fc086d0db479d3a3a9cd210ee5fabde1a0a0d606/SQLite%20Tests/ExpressionTests.swift#L397-L399

**Comment by stephencelis** [comment]
P.S. You can call `.first` on a query to automatically add `LIMIT 1` and return the row itself:

``` swift
let result = table.filter(exp.difficulty == difficulty.rawValue).order(random()).first!
```

**Comment by acarr** [comment]
Even then it still says it's ambiguous. Would the way the framework is linked have anything to do with this? I believe I followed those instructions in the getting started. 

![screen shot 2015-04-10 at 11 45 04 am](https://cloud.githubusercontent.com/assets/237203/7091370/35b320d2-df77-11e4-9884-d45bd02b16a2.png)

Also, thanks for the `.first` note!

**Comment by stephencelis** [comment]
If you remove the `order` clause does everything still work? Chaining functions in Swift can sometimes yield an unrelated error. Try, _e.g._, the following and see if your error changes:

``` swift
var result = table.filter(exp.difficulty == difficulty.rawValue)
result = result.order(random())
result = result.limit(1)
```

**Comment by acarr** [comment]
Without `order` everything else still works. That's how I have it now to get it to build.

Changing it into separate statements still throws an error just now at the second line. Also tried it with and without the module name.

**Comment by stephencelis** [comment]
I'm not sure what it could be ambiguous with in its own module. Have you cleaned your build folder (trashed Derived Data)?

If all else fails you can use `Expression<Int>(literal: "random()")`.

**Comment by stephencelis** [comment]
@acarr This appears to be a Swift bug. I've filed rdar://20513886 ("Swift regression: "ambiguous use of funcname" where there is no ambiguity"). Closing this for now as I can't fix it without renaming the function :/

Casting is probably the best workaround till Apple fixes this.

``` swift
order(random() as Expression<Int>)
```

I may update `random()` to return an `Expression<Int64>` in the near future. I also may rename it entirely. So...head's up :)
